sunpy/map/mapbase.py

```diff
@@ -1902,6 +1908,66 @@ def draw_limb(self, axes=None, **kwargs):
 
         return [circ]
 
+    @u.quantity_input
+    def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = None,
+                        axes=None, top_right=None, **kwargs):
+        """
+        Draw a quadrangle defined in world coordinates on the plot using Astropy's
+        `~astropy.visualization.wcsaxes.patches.Quadrangle`.
+
+        This draws a quadrangle that has corners at ``(bottom_left, top_right)``,
+        and has sides aligned with the coordinate axes of the frame of ``bottom_left``,
+        which may be different from the coordinate axes of the map.
+
+        If ``width`` and ``height`` are specified, they are respectively added to the
+        longitude and latitude of the ``bottom_left`` coordinate to calculate a
+        ``top_right`` coordinate.
+
+        Parameters
+        ----------
+        bottom_left : `~astropy.coordinates.SkyCoord`
+            The bottom-left coordinate of the quadrangle. It can
+            have shape ``(2,)`` to simultaneously define ``top_right``.
+        top_right : `~astropy.coordinates.SkyCoord`
+            The top-right coordinate of the quadrangle.
+        width : `astropy.units.Quantity`, optional
+            The width of the quadrangle. Required if ``top_right`` is omitted.
+        height : `astropy.units.Quantity`
+            The height of the quadrangle. Required if ``top_right`` is omitted.
+        axes : `matplotlib.axes.Axes`
+            The axes on which to plot the quadrangle. Defaults to the current
+            axes.
+
+        Returns
+        -------
+        quad : `~astropy.visualization.wcsaxes.patches.Quadrangle`
+            The added patch
+
+        Notes
+        -----
+        Extra keyword arguments to this function are passed through to the
+        `~astropy.visualization.wcsaxes.patches.Quadrangle` instance.
+        """
+        bottom_left, top_right = get_rectangle_coordinates(
+            bottom_left, top_right=top_right, width=width, height=height)
+
+        width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
+        height = top_right.spherical.lat - bottom_left.spherical.lat
+
+        if not axes:
+            axes = plt.gca()
```

Comment:
Am I right in thinking this will not work if we are not using WCSAxes? if so we should use the wcsaxes helper and check that it is an instance of wcsaxes afterwards?

Reply:
Yes I was under the assumption that this would be used only with WCSAxes, if not should I include a check such as ```is_wcsaxes(axes)``` and raise an error if it's not WCSAxes?

Reply:
I added a check (and a test)